plugins/pulp_rpm/plugins/importers/iso/importer.py

```diff
@@ -129,10 +127,14 @@ def upload_unit(self, transfer_repo, type_id, unit_key, metadata, file_path, con
         :param config: plugin configuration for the repository
         :type  config: pulp.plugins.config.PluginCallConfiguration
         """
-        if models.ISO.objects.filter(**unit_key).count() > 0:
-            return {'success_flag': False, 'summary': _('ISO already exists'), 'details': None}
+        qs = models.ISO.objects.filter(**unit_key)
+        if qs:
+            # iso with this key already exists, use it
+            iso = qs.first()
```

Comment:
This if-else block looks a bit redundant to me because later in [try-except](https://github.com/pulp/pulp_rpm/pull/1033/files#diff-a1402e54dd26b2332344c553bf5fa30aR149) we will know if ISO already exists or not and the old one will be chosen in case of the duplicates.
My other concern is that we choose old ISO but then [validate the recently uploaded one](https://github.com/pulp/pulp_rpm/pull/1033/files#diff-a1402e54dd26b2332344c553bf5fa30aR143), it probably does not make sense since we won't use it at all.

My suggestion is to let try-except decide which ISO to keep.
If there is a reason to do an early check on existing ISO, then skip validation of the uploaded ISO if it won't be imported.